Unsupported class file major version 65

Using Forgix on a MC 1.20.5 project fails
```
> Task :mergeJars FAILED

Execution failed for task ':mergeJars'.
> Unsupported class file major version 65

* Try:
> Run with --info or --debug option to get more log output.
> Run with --scan to get full insights.
> Get more help at https://help.gradle.org.

* Exception is:
org.gradle.api.tasks.TaskExecutionException: Execution failed for task ':mergeJars'.
	at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.lambda$executeIfValid$1(ExecuteActionsTaskExecuter.java:130)
	at org.gradle.internal.Try$Failure.ifSuccessfulOrElse(Try.java:282)
	at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.executeIfValid(ExecuteActionsTaskExecuter.java:128)
	at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.execute(ExecuteActionsTaskExecuter.java:116)
	at org.gradle.api.internal.tasks.execution.FinalizePropertiesTaskExecuter.execute(FinalizePropertiesTaskExecuter.java:46)
	at org.gradle.api.internal.tasks.execution.ResolveTaskExecutionModeExecuter.execute(ResolveTaskExecutionModeExecuter.java:51)
	at org.gradle.api.internal.tasks.execution.SkipTaskWithNoActionsExecuter.execute(SkipTaskWithNoActionsExecuter.java:57)
	at org.gradle.api.internal.tasks.execution.SkipOnlyIfTaskExecuter.execute(SkipOnlyIfTaskExecuter.java:74)
	at org.gradle.api.internal.tasks.execution.CatchExceptionTaskExecuter.execute(CatchExceptionTaskExecuter.java:36)
	at org.gradle.api.internal.tasks.execution.EventFiringTaskExecuter$1.executeTask(EventFiringTaskExecuter.java:77)
	at org.gradle.api.internal.tasks.execution.EventFiringTaskExecuter$1.call(EventFiringTaskExecuter.java:55)
	at org.gradle.api.internal.tasks.execution.EventFiringTaskExecuter$1.call(EventFiringTaskExecuter.java:52)
	at org.gradle.internal.operations.DefaultBuildOperationRunner$CallableBuildOperationWorker.execute(DefaultBuildOperationRunner.java:200)
	at org.gradle.internal.operations.DefaultBuildOperationRunner$CallableBuildOperationWorker.execute(DefaultBuildOperationRunner.java:195)
	at org.gradle.internal.operations.DefaultBuildOperationRunner$2.execute(DefaultBuildOperationRunner.java:66)
	at org.gradle.internal.operations.DefaultBuildOperationRunner$2.execute(DefaultBuildOperationRunner.java:59)
	at org.gradle.internal.operations.DefaultBuildOperationRunner.execute(DefaultBuildOperationRunner.java:157)
	at org.gradle.internal.operations.DefaultBuildOperationRunner.execute(DefaultBuildOperationRunner.java:59)
	at org.gradle.internal.operations.DefaultBuildOperationRunner.call(DefaultBuildOperationRunner.java:53)
	at org.gradle.internal.operations.DefaultBuildOperationExecutor.call(DefaultBuildOperationExecutor.java:73)
	at org.gradle.api.internal.tasks.execution.EventFiringTaskExecuter.execute(EventFiringTaskExecuter.java:52)
	at org.gradle.execution.plan.LocalTaskNodeExecutor.execute(LocalTaskNodeExecutor.java:42)
	at org.gradle.execution.taskgraph.DefaultTaskExecutionGraph$InvokeNodeExecutorsAction.execute(DefaultTaskExecutionGraph.java:331)
	at org.gradle.execution.taskgraph.DefaultTaskExecutionGraph$InvokeNodeExecutorsAction.execute(DefaultTaskExecutionGraph.java:318)
	at org.gradle.execution.taskgraph.DefaultTaskExecutionGraph$BuildOperationAwareExecutionAction.lambda$execute$0(DefaultTaskExecutionGraph.java:314)
	at org.gradle.internal.operations.CurrentBuildOperationRef.with(CurrentBuildOperationRef.java:80)
	at org.gradle.execution.taskgraph.DefaultTaskExecutionGraph$BuildOperationAwareExecutionAction.execute(DefaultTaskExecutionGraph.java:314)
	at org.gradle.execution.taskgraph.DefaultTaskExecutionGraph$BuildOperationAwareExecutionAction.execute(DefaultTaskExecutionGraph.java:303)
	at org.gradle.execution.plan.DefaultPlanExecutor$ExecutorWorker.execute(DefaultPlanExecutor.java:463)
	at org.gradle.execution.plan.DefaultPlanExecutor$ExecutorWorker.run(DefaultPlanExecutor.java:380)
	at org.gradle.internal.concurrent.ExecutorPolicy$CatchAndRecordFailures.onExecute(ExecutorPolicy.java:64)
	at org.gradle.internal.concurrent.AbstractManagedExecutor$1.run(AbstractManagedExecutor.java:47)
Caused by: java.lang.IllegalArgumentException: Unsupported class file major version 65
	at pacifistmc.libraries.org.objectweb.asm.ClassReader.<init>(ClassReader.java:199)
	at pacifistmc.libraries.org.objectweb.asm.ClassReader.<init>(ClassReader.java:180)
	at pacifistmc.libraries.org.objectweb.asm.ClassReader.<init>(ClassReader.java:166)
	at pacifistmc.libraries.org.objectweb.asm.ClassReader.<init>(ClassReader.java:287)
	at pacifistmc.libraries.me.lucko.jarrelocator.JarRelocatorTask.processClass(JarRelocatorTask.java:164)
	at pacifistmc.libraries.me.lucko.jarrelocator.JarRelocatorTask.processEntry(JarRelocatorTask.java:100)
	at pacifistmc.libraries.me.lucko.jarrelocator.JarRelocatorTask.processEntries(JarRelocatorTask.java:87)
	at pacifistmc.libraries.me.lucko.jarrelocator.JarRelocator.run(JarRelocator.java:89)
	at io.github.pacifistmc.forgix.Forgix$Merge.remap(Forgix.java:402)
	at io.github.pacifistmc.forgix.Forgix$Merge.merge(Forgix.java:133)
	at io.github.pacifistmc.forgix.plugin.MergeJarsTask.mergeJars(MergeJarsTask.java:144)
	at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103)
	at org.gradle.internal.reflect.JavaMethod.invoke(JavaMethod.java:125)
	at org.gradle.api.internal.project.taskfactory.StandardTaskAction.doExecute(StandardTaskAction.java:58)
	at org.gradle.api.internal.project.taskfactory.StandardTaskAction.execute(StandardTaskAction.java:51)
	at org.gradle.api.internal.project.taskfactory.StandardTaskAction.execute(StandardTaskAction.java:29)
	at org.gradle.api.internal.tasks.execution.TaskExecution$3.run(TaskExecution.java:244)
	at org.gradle.internal.operations.DefaultBuildOperationRunner$1.execute(DefaultBuildOperationRunner.java:29)
	at org.gradle.internal.operations.DefaultBuildOperationRunner$1.execute(DefaultBuildOperationRunner.java:26)
	at org.gradle.internal.operations.DefaultBuildOperationRunner$2.execute(DefaultBuildOperationRunner.java:66)
	at org.gradle.internal.operations.DefaultBuildOperationRunner$2.execute(DefaultBuildOperationRunner.java:59)
	at org.gradle.internal.operations.DefaultBuildOperationRunner.execute(DefaultBuildOperationRunner.java:157)
	at org.gradle.internal.operations.DefaultBuildOperationRunner.execute(DefaultBuildOperationRunner.java:59)
	at org.gradle.internal.operations.DefaultBuildOperationRunner.run(DefaultBuildOperationRunner.java:47)
	at org.gradle.internal.operations.DefaultBuildOperationExecutor.run(DefaultBuildOperationExecutor.java:68)
	at org.gradle.api.internal.tasks.execution.TaskExecution.executeAction(TaskExecution.java:229)
	at org.gradle.api.internal.tasks.execution.TaskExecution.executeActions(TaskExecution.java:212)
	at org.gradle.api.internal.tasks.execution.TaskExecution.executeWithPreviousOutputFiles(TaskExecution.java:195)
	at org.gradle.api.internal.tasks.execution.TaskExecution.execute(TaskExecution.java:162)
	at org.gradle.internal.execution.steps.ExecuteStep.executeInternal(ExecuteStep.java:105)
	at org.gradle.internal.execution.steps.ExecuteStep.access$000(ExecuteStep.java:44)
	at org.gradle.internal.execution.steps.ExecuteStep$1.call(ExecuteStep.java:59)
	at org.gradle.internal.execution.steps.ExecuteStep$1.call(ExecuteStep.java:56)
	at org.gradle.internal.operations.DefaultBuildOperationRunner$CallableBuildOperationWorker.execute(DefaultBuildOperationRunner.java:200)
	at org.gradle.internal.operations.DefaultBuildOperationRunner$CallableBuildOperationWorker.execute(DefaultBuildOperationRunner.java:195)
	at org.gradle.internal.operations.DefaultBuildOperationRunner$2.execute(DefaultBuildOperationRunner.java:66)
	at org.gradle.internal.operations.DefaultBuildOperationRunner$2.execute(DefaultBuildOperationRunner.java:59)
	at org.gradle.internal.operations.DefaultBuildOperationRunner.execute(DefaultBuildOperationRunner.java:157)
	at org.gradle.internal.operations.DefaultBuildOperationRunner.execute(DefaultBuildOperationRunner.java:59)
	at org.gradle.internal.operations.DefaultBuildOperationRunner.call(DefaultBuildOperationRunner.java:53)
	at org.gradle.internal.operations.DefaultBuildOperationExecutor.call(DefaultBuildOperationExecutor.java:73)
	at org.gradle.internal.execution.steps.ExecuteStep.execute(ExecuteStep.java:56)
	at org.gradle.internal.execution.steps.ExecuteStep.execute(ExecuteStep.java:44)
	at org.gradle.internal.execution.steps.CancelExecutionStep.execute(CancelExecutionStep.java:41)
	at org.gradle.internal.execution.steps.TimeoutStep.executeWithoutTimeout(TimeoutStep.java:74)
	at org.gradle.internal.execution.steps.TimeoutStep.execute(TimeoutStep.java:55)
	at org.gradle.internal.execution.steps.PreCreateOutputParentsStep.execute(PreCreateOutputParentsStep.java:50)
	at org.gradle.internal.execution.steps.PreCreateOutputParentsStep.execute(PreCreateOutputParentsStep.java:28)
	at org.gradle.internal.execution.steps.RemovePreviousOutputsStep.execute(RemovePreviousOutputsStep.java:67)
	at org.gradle.internal.execution.steps.RemovePreviousOutputsStep.execute(RemovePreviousOutputsStep.java:37)
	at org.gradle.internal.execution.steps.BroadcastChangingOutputsStep.execute(BroadcastChangingOutputsStep.java:61)
	at org.gradle.internal.execution.steps.BroadcastChangingOutputsStep.execute(BroadcastChangingOutputsStep.java:26)
	at org.gradle.internal.execution.steps.CaptureOutputsAfterExecutionStep.execute(CaptureOutputsAfterExecutionStep.java:67)
	at org.gradle.internal.execution.steps.CaptureOutputsAfterExecutionStep.execute(CaptureOutputsAfterExecutionStep.java:45)
	at org.gradle.internal.execution.steps.ResolveInputChangesStep.execute(ResolveInputChangesStep.java:40)
	at org.gradle.internal.execution.steps.ResolveInputChangesStep.execute(ResolveInputChangesStep.java:29)
	at org.gradle.internal.execution.steps.BuildCacheStep.executeWithoutCache(BuildCacheStep.java:189)
	at org.gradle.internal.execution.steps.BuildCacheStep.lambda$execute$1(BuildCacheStep.java:75)
	at org.gradle.internal.Either$Right.fold(Either.java:175)
	at org.gradle.internal.execution.caching.CachingState.fold(CachingState.java:62)
	at org.gradle.internal.execution.steps.BuildCacheStep.execute(BuildCacheStep.java:73)
	at org.gradle.internal.execution.steps.BuildCacheStep.execute(BuildCacheStep.java:48)
	at org.gradle.internal.execution.steps.StoreExecutionStateStep.execute(StoreExecutionStateStep.java:46)
	at org.gradle.internal.execution.steps.StoreExecutionStateStep.execute(StoreExecutionStateStep.java:35)
	at org.gradle.internal.execution.steps.SkipUpToDateStep.executeBecause(SkipUpToDateStep.java:76)
	at org.gradle.internal.execution.steps.SkipUpToDateStep.lambda$execute$2(SkipUpToDateStep.java:54)
	at org.gradle.internal.execution.steps.SkipUpToDateStep.execute(SkipUpToDateStep.java:54)
	at org.gradle.internal.execution.steps.SkipUpToDateStep.execute(SkipUpToDateStep.java:36)
	at org.gradle.internal.execution.steps.legacy.MarkSnapshottingInputsFinishedStep.execute(MarkSnapshottingInputsFinishedStep.java:37)
	at org.gradle.internal.execution.steps.legacy.MarkSnapshottingInputsFinishedStep.execute(MarkSnapshottingInputsFinishedStep.java:27)
	at org.gradle.internal.execution.steps.ResolveIncrementalCachingStateStep.executeDelegate(ResolveIncrementalCachingStateStep.java:49)
	at org.gradle.internal.execution.steps.ResolveIncrementalCachingStateStep.executeDelegate(ResolveIncrementalCachingStateStep.java:27)
	at org.gradle.internal.execution.steps.AbstractResolveCachingStateStep.execute(AbstractResolveCachingStateStep.java:71)
	at org.gradle.internal.execution.steps.AbstractResolveCachingStateStep.execute(AbstractResolveCachingStateStep.java:39)
	at org.gradle.internal.execution.steps.ResolveChangesStep.execute(ResolveChangesStep.java:65)
	at org.gradle.internal.execution.steps.ResolveChangesStep.execute(ResolveChangesStep.java:36)
	at org.gradle.internal.execution.steps.ValidateStep.execute(ValidateStep.java:106)
	at org.gradle.internal.execution.steps.ValidateStep.execute(ValidateStep.java:55)
	at org.gradle.internal.execution.steps.AbstractCaptureStateBeforeExecutionStep.execute(AbstractCaptureStateBeforeExecutionStep.java:64)
	at org.gradle.internal.execution.steps.AbstractCaptureStateBeforeExecutionStep.execute(AbstractCaptureStateBeforeExecutionStep.java:43)
	at org.gradle.internal.execution.steps.AbstractSkipEmptyWorkStep.executeWithNonEmptySources(AbstractSkipEmptyWorkStep.java:125)
	at org.gradle.internal.execution.steps.AbstractSkipEmptyWorkStep.execute(AbstractSkipEmptyWorkStep.java:56)
	at org.gradle.internal.execution.steps.AbstractSkipEmptyWorkStep.execute(AbstractSkipEmptyWorkStep.java:36)
	at org.gradle.internal.execution.steps.legacy.MarkSnapshottingInputsStartedStep.execute(MarkSnapshottingInputsStartedStep.java:38)
	at org.gradle.internal.execution.steps.LoadPreviousExecutionStateStep.execute(LoadPreviousExecutionStateStep.java:36)
	at org.gradle.internal.execution.steps.LoadPreviousExecutionStateStep.execute(LoadPreviousExecutionStateStep.java:23)
	at org.gradle.internal.execution.steps.HandleStaleOutputsStep.execute(HandleStaleOutputsStep.java:75)
	at org.gradle.internal.execution.steps.HandleStaleOutputsStep.execute(HandleStaleOutputsStep.java:41)
	at org.gradle.internal.execution.steps.AssignMutableWorkspaceStep.lambda$execute$0(AssignMutableWorkspaceStep.java:35)
	at org.gradle.api.internal.tasks.execution.TaskExecution$4.withWorkspace(TaskExecution.java:289)
	at org.gradle.internal.execution.steps.AssignMutableWorkspaceStep.execute(AssignMutableWorkspaceStep.java:31)
	at org.gradle.internal.execution.steps.AssignMutableWorkspaceStep.execute(AssignMutableWorkspaceStep.java:22)
	at org.gradle.internal.execution.steps.ChoosePipelineStep.execute(ChoosePipelineStep.java:40)
	at org.gradle.internal.execution.steps.ChoosePipelineStep.execute(ChoosePipelineStep.java:23)
	at org.gradle.internal.execution.steps.ExecuteWorkBuildOperationFiringStep.lambda$execute$2(ExecuteWorkBuildOperationFiringStep.java:67)
	at org.gradle.internal.execution.steps.ExecuteWorkBuildOperationFiringStep.execute(ExecuteWorkBuildOperationFiringStep.java:67)
	at org.gradle.internal.execution.steps.ExecuteWorkBuildOperationFiringStep.execute(ExecuteWorkBuildOperationFiringStep.java:39)
	at org.gradle.internal.execution.steps.IdentityCacheStep.execute(IdentityCacheStep.java:46)
	at org.gradle.internal.execution.steps.IdentityCacheStep.execute(IdentityCacheStep.java:34)
	at org.gradle.internal.execution.steps.IdentifyStep.execute(IdentifyStep.java:48)
	at org.gradle.internal.execution.steps.IdentifyStep.execute(IdentifyStep.java:35)
	at org.gradle.internal.execution.impl.DefaultExecutionEngine$1.execute(DefaultExecutionEngine.java:61)
	at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.executeIfValid(ExecuteActionsTaskExecuter.java:127)
	at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.execute(ExecuteActionsTaskExecuter.java:116)
	at org.gradle.api.internal.tasks.execution.FinalizePropertiesTaskExecuter.execute(FinalizePropertiesTaskExecuter.java:46)
	at org.gradle.api.internal.tasks.execution.ResolveTaskExecutionModeExecuter.execute(ResolveTaskExecutionModeExecuter.java:51)
	at org.gradle.api.internal.tasks.execution.SkipTaskWithNoActionsExecuter.execute(SkipTaskWithNoActionsExecuter.java:57)
	at org.gradle.api.internal.tasks.execution.SkipOnlyIfTaskExecuter.execute(SkipOnlyIfTaskExecuter.java:74)
	at org.gradle.api.internal.tasks.execution.CatchExceptionTaskExecuter.execute(CatchExceptionTaskExecuter.java:36)
	at org.gradle.api.internal.tasks.execution.EventFiringTaskExecuter$1.executeTask(EventFiringTaskExecuter.java:77)
	at org.gradle.api.internal.tasks.execution.EventFiringTaskExecuter$1.call(EventFiringTaskExecuter.java:55)
	at org.gradle.api.internal.tasks.execution.EventFiringTaskExecuter$1.call(EventFiringTaskExecuter.java:52)
	at org.gradle.internal.operations.DefaultBuildOperationRunner$CallableBuildOperationWorker.execute(DefaultBuildOperationRunner.java:200)
	at org.gradle.internal.operations.DefaultBuildOperationRunner$CallableBuildOperationWorker.execute(DefaultBuildOperationRunner.java:195)
	at org.gradle.internal.operations.DefaultBuildOperationRunner$2.execute(DefaultBuildOperationRunner.java:66)
	at org.gradle.internal.operations.DefaultBuildOperationRunner$2.execute(DefaultBuildOperationRunner.java:59)
	at org.gradle.internal.operations.DefaultBuildOperationRunner.execute(DefaultBuildOperationRunner.java:157)
	at org.gradle.internal.operations.DefaultBuildOperationRunner.execute(DefaultBuildOperationRunner.java:59)
	at org.gradle.internal.operations.DefaultBuildOperationRunner.call(DefaultBuildOperationRunner.java:53)
	at org.gradle.internal.operations.DefaultBuildOperationExecutor.call(DefaultBuildOperationExecutor.java:73)
	at org.gradle.api.internal.tasks.execution.EventFiringTaskExecuter.execute(EventFiringTaskExecuter.java:52)
	at org.gradle.execution.plan.LocalTaskNodeExecutor.execute(LocalTaskNodeExecutor.java:42)
	at org.gradle.execution.taskgraph.DefaultTaskExecutionGraph$InvokeNodeExecutorsAction.execute(DefaultTaskExecutionGraph.java:331)
	at org.gradle.execution.taskgraph.DefaultTaskExecutionGraph$InvokeNodeExecutorsAction.execute(DefaultTaskExecutionGraph.java:318)
	at org.gradle.execution.taskgraph.DefaultTaskExecutionGraph$BuildOperationAwareExecutionAction.lambda$execute$0(DefaultTaskExecutionGraph.java:314)
	at org.gradle.internal.operations.CurrentBuildOperationRef.with(CurrentBuildOperationRef.java:80)
	at org.gradle.execution.taskgraph.DefaultTaskExecutionGraph$BuildOperationAwareExecutionAction.execute(DefaultTaskExecutionGraph.java:314)
	at org.gradle.execution.taskgraph.DefaultTaskExecutionGraph$BuildOperationAwareExecutionAction.execute(DefaultTaskExecutionGraph.java:303)
	at org.gradle.execution.plan.DefaultPlanExecutor$ExecutorWorker.execute(DefaultPlanExecutor.java:463)
	at org.gradle.execution.plan.DefaultPlanExecutor$ExecutorWorker.run(DefaultPlanExecutor.java:380)
	at org.gradle.internal.concurrent.ExecutorPolicy$CatchAndRecordFailures.onExecute(ExecutorPolicy.java:64)
	at org.gradle.internal.concurrent.AbstractManagedExecutor$1.run(AbstractManagedExecutor.java:47)

```

--- a/README.md
+++ b/README.md
@@ -152,6 +152,18 @@ subprojects {
   - This exists because Forge can be a real pain at times, and Forge sometimes does something strange where we can’t actually identify mixins the normal way. However, if we don’t automatically detect the mixins, then only this should be used to specify the mixins explicitly.
   - This can be used more than once to specify multiple mixins.
 
+##### NeoForge sub-container (“neoforge”)
+- `projectName` (String)
+  - This is the name of the Forge project. This is set to “neoforge” by default.
+- `jarLocation` (String)
+  - This is the location of the built NeoForge jar **from the project that’s specified in `projectName`**. By default, this retrieves the jar with the shortest name, which is quite scuffed but I don’t know how to retrieve the built jar without relying on loom or something similar, hopefully it’ll be better in the future though!
+- `additionalRelocate` (String, String)
+  - Simply put, this allows you to define more `group`s, which is useful for relocating libraries.
+  - This can be used numerous times to specify multiple relocations.
+- `mixin` (String)
+  - This exists because NeoForge can be a real pain at times, and NeoForge sometimes does something strange where we can’t actually identify mixins the normal way. However, if we don’t automatically detect the mixins, then only this should be used to specify the mixins explicitly.
+  - This can be used more than once to specify multiple mixins.
+
 ##### Quilt sub-container (“quilt”)
 - `projectName` (String)
   - This is the name of the Quilt project. This is set to “quilt” by default.
@@ -198,6 +210,17 @@ forgix {
         
         mixin "forge.mixins.json"
         mixin "forge.mixins.another.json"
+    }
+
+    neoforge {
+      projectName = "neoforge"
+      jarLocation = "build/libs/example-mod.jar"
+  
+      additionalRelocate "org.my.lib" "neoforge.org.my.lib"
+      additionalRelocate "org.my.lib.another" "neoforge.org.my.lib.another"
+  
+      mixin "neoforge.mixins.json"
+      mixin "neoforge.mixins.another.json"
     }
     
     fabric {

--- a/build.gradle
+++ b/build.gradle
@@ -1,13 +1,13 @@
 plugins {
     id 'java'
     id 'java-gradle-plugin'
-    id 'com.github.johnrengelman.shadow' version '7.0.0'
+    id 'com.github.johnrengelman.shadow' version '8.0.0'
     id 'maven-publish'
-    id 'com.gradle.plugin-publish' version '0.16.0'
+    id 'com.gradle.plugin-publish' version '0.21.0'
 }
 
 group 'io.github.pacifistmc.forgix'
-version '1.2.6' // Make sure to change the version number in io.github.pacifistmc.forgix.Forgix
+version '1.2.7' // Make sure to change the version number in io.github.pacifistmc.forgix.Forgix
 
 configurations {
     shadowMe
@@ -21,15 +21,15 @@ repositories {
 dependencies {
     implementation gradleApi()
 
-    shadowMe 'commons-io:commons-io:2.11.0'
+    shadowMe 'commons-io:commons-io:2.16.1'
 
     // Magic
-    shadowMe('me.lucko:jar-relocator:1.5') {
+    shadowMe('me.lucko:jar-relocator:1.7') {
         exclude group: 'org.ow2.asm'
     }
-    shadowMe 'org.ow2.asm:asm:9.3'
-    shadowMe 'org.ow2.asm:asm-commons:9.3'
-    shadowMe 'net.lingala.zip4j:zip4j:2.11.1'
+    shadowMe 'org.ow2.asm:asm:9.7'
+    shadowMe 'org.ow2.asm:asm-commons:9.7'
+    shadowMe 'net.lingala.zip4j:zip4j:2.11.5'
 
     shadowMe fileTree(dir: 'libs', include: ['*.jar']) // https://bitbucket.org/ReaperSoon/maven-repo/src/367c0bcb2749/fr/stevecohen/jarmanager
 }

--- a/gradle/wrapper/gradle-wrapper.properties
+++ b/gradle/wrapper/gradle-wrapper.properties
@@ -1,5 +1,5 @@
 distributionBase=GRADLE_USER_HOME
 distributionPath=wrapper/dists
-distributionUrl=https\://services.gradle.org/distributions/gradle-7.4-bin.zip
+distributionUrl=https\://services.gradle.org/distributions/gradle-8.4-bin.zip
 zipStoreBase=GRADLE_USER_HOME
 zipStorePath=wrapper/dists

--- a/src/main/java/io/github/pacifistmc/forgix/Forgix.java
+++ b/src/main/java/io/github/pacifistmc/forgix/Forgix.java
@@ -34,7 +34,7 @@ public class Forgix {
     public static final String manifestVersionKey = "Forgix-Version";
 
     public static class Merge {
-        private final String version = "1.2.6";
+        private final String version = "1.2.7";
         public static Set<PosixFilePermission> perms;
 
         static {

--- a/src/main/java/io/github/pacifistmc/forgix/Forgix.java
+++ b/src/main/java/io/github/pacifistmc/forgix/Forgix.java
@@ -53,6 +53,9 @@ public class Forgix {
         private File forgeJar;
         private Map<String, String> forgeRelocations;
         private List<String> forgeMixins;
+        private File neoforgeJar;
+        private Map<String, String> neoforgeRelocations;
+        private List<String> neoforgeMixins;
         private File fabricJar;
         private Map<String, String> fabricRelocations;
         private File quiltJar;
@@ -67,10 +70,13 @@ public class Forgix {
         private final Logger logger;
         private final Map<String, String> removeDuplicateRelocations = new HashMap<>();
 
-        public Merge(@Nullable File forgeJar, Map<String, String> forgeRelocations, List<String> forgeMixins, @Nullable File fabricJar, Map<String, String> fabricRelocations, @Nullable File quiltJar, Map<String, String> quiltRelocations, Map<ForgixMergeExtension.CustomContainer, File> customContainerMap, String group, File tempDir, String mergedJarName, List<String> removeDuplicates, Logger logger) {
+        public Merge(@Nullable File forgeJar, Map<String, String> forgeRelocations, List<String> forgeMixins, @Nullable File neoforgeJar, Map<String, String> neoforgeRelocations, List<String> neoforgeMixins, @Nullable File fabricJar, Map<String, String> fabricRelocations, @Nullable File quiltJar, Map<String, String> quiltRelocations, Map<ForgixMergeExtension.CustomContainer, File> customContainerMap, String group, File tempDir, String mergedJarName, List<String> removeDuplicates, Logger logger) {
             this.forgeJar = forgeJar;
             this.forgeRelocations = forgeRelocations;
             this.forgeMixins = forgeMixins;
+            this.neoforgeJar = neoforgeJar;
+            this.neoforgeRelocations = neoforgeRelocations;
+            this.neoforgeMixins = neoforgeMixins;
             this.fabricJar = fabricJar;
             this.fabricRelocations = fabricRelocations;
             this.quiltJar = quiltJar;
@@ -97,12 +103,16 @@ public class Forgix {
             }
 
             tempDir.mkdirs();
-            if (forgeJar == null && fabricJar == null && quiltJar == null && customContainerMap.isEmpty()) {
+            if (forgeJar == null && neoforgeJar == null && fabricJar == null && quiltJar == null && customContainerMap.isEmpty()) {
                 throw new IllegalArgumentException("No jars were provided.");
             }
 
             if (forgeJar != null && !forgeJar.exists()) {
                 logger.warn("Forge jar does not exist! You can ignore this if you are not using forge.\nYou might want to change Forgix settings if something is wrong.");
+            }
+
+            if (neoforgeJar != null && !neoforgeJar.exists()) {
+                logger.warn("NeoForge jar does not exist! You can ignore this if you are not using neoforge.\nYou might want to change Forgix settings if something is wrong.");
             }
 
             if (fabricJar != null && !fabricJar.exists()) {
@@ -123,6 +133,7 @@ public class Forgix {
 
             logger.info("\nSettings:\n" +
                     "Forge: " + (forgeJar == null || !forgeJar.exists() ? "No\n" : "Yes\n") +
+                    "NeoForge: " + (neoforgeJar == null || !neoforgeJar.exists() ? "No\n" : "Yes\n") +
                     "Fabric: " + (fabricJar == null || !fabricJar.exists() ? "No\n" : "Yes\n") +
                     "Quilt: " + (quiltJar == null || !quiltJar.exists() ? "No\n" : "Yes\n") +
                     "Custom Containers: " + (customContainerMap.isEmpty() ? "No\n" : "Yes\n") +
@@ -134,6 +145,7 @@ public class Forgix {
 
             File fabricTemps = new File(tempDir, "fabric-temps");
             File forgeTemps = new File(tempDir, "forge-temps");
+            File neoforgeTemps = new File(tempDir, "neoforge-temps");
             File quiltTemps = new File(tempDir, "quilt-temps");
 
             customContainerTemps = new HashMap<>();
@@ -150,6 +162,9 @@ public class Forgix {
             if (forgeTemps.exists()) FileUtils.deleteQuietly(forgeTemps);
             forgeTemps.mkdirs();
 
+            if (neoforgeTemps.exists()) FileUtils.deleteQuietly(neoforgeTemps);
+            neoforgeTemps.mkdirs();
+
             if (quiltTemps.exists()) FileUtils.deleteQuietly(quiltTemps);
             quiltTemps.mkdirs();
 
@@ -162,6 +177,7 @@ public class Forgix {
 
             JarUnpacker jarUnpacker = new JarUnpacker();
             if (forgeJar != null && forgeJar.exists()) jarUnpacker.unpack(forgeJar.getAbsolutePath(), forgeTemps.getAbsolutePath());
+            if (neoforgeJar != null && neoforgeJar.exists()) jarUnpacker.unpack(neoforgeJar.getAbsolutePath(), neoforgeTemps.getAbsolutePath());
             if (fabricJar != null && fabricJar.exists()) jarUnpacker.unpack(fabricJar.getAbsolutePath(), fabricTemps.getAbsolutePath());
             if (quiltJar != null && quiltJar.exists()) jarUnpacker.unpack(quiltJar.getAbsolutePath(), quiltTemps.getAbsolutePath());
 
@@ -177,12 +193,15 @@ public class Forgix {
 
             Manifest mergedManifest = new Manifest();
             Manifest forgeManifest = new Manifest();
+            Manifest neoforgeManifest = new Manifest();
             Manifest fabricManifest = new Manifest();
             Manifest quiltManifest = new Manifest();
             List<Manifest> customContainerManifests = new ArrayList<>();
 
             FileInputStream fileInputStream = null;
             if (forgeJar != null && forgeJar.exists()) forgeManifest.read(fileInputStream = new FileInputStream(new File(forgeTemps, "META-INF/MANIFEST.MF")));
+            if (fileInputStream != null) fileInputStream.close();
+            if (neoforgeJar != null && neoforgeJar.exists()) neoforgeManifest.read(fileInputStream = new FileInputStream(new File(neoforgeTemps, "META-INF/MANIFEST.MF")));
             if (fileInputStream != null) fileInputStream.close();
             if (fabricJar != null && fabricJar.exists()) fabricManifest.read(fileInputStream = new FileInputStream(new File(fabricTemps, "META-INF/MANIFEST.MF")));
             if (fileInputStream != null) fileInputStream.close();
@@ -201,6 +220,7 @@ public class Forgix {
             }
 
             forgeManifest.getMainAttributes().forEach((key, value) -> mergedManifest.getMainAttributes().putValue(key.toString(), value.toString()));
+            neoforgeManifest.getMainAttributes().forEach((key, value) -> mergedManifest.getMainAttributes().putValue(key.toString(), value.toString()));
             fabricManifest.getMainAttributes().forEach((key, value) -> mergedManifest.getMainAttributes().putValue(key.toString(), value.toString()));
             quiltManifest.getMainAttributes().forEach((key, value) -> mergedManifest.getMainAttributes().putValue(key.toString(), value.toString()));
 
@@ -220,7 +240,11 @@ public class Forgix {
                 }
 
                 for (String mixin : mixins) {
-                    remappedMixin.add("forge-" + mixin);
+                    if (mixin.contains("neoforge") || mixin.contains("neo")) {
+                        remappedMixin.add("neoforge-" + mixin);
+                    } else {
+                        remappedMixin.add("forge-" + mixin);
+                    }
                 }
 
                 mergedManifest.getMainAttributes().putValue("MixinConfigs", String.join(",", remappedMixin));
@@ -235,7 +259,16 @@ public class Forgix {
                 if (!forgeMixins.isEmpty()) mergedManifest.getMainAttributes().putValue("MixinConfigs", String.join(",", this.forgeMixins));
             }
 
-            remapResources(forgeTemps, fabricTemps, quiltTemps);
+            if (this.neoforgeMixins != null) {
+                List<String> newNeoForgeMixins = new ArrayList<>();
+                for (String mixin : this.neoforgeMixins) {
+                    newNeoForgeMixins.add("neoforge-" + mixin);
+                }
+                this.neoforgeMixins = newNeoForgeMixins;
+                if (!neoforgeMixins.isEmpty()) mergedManifest.getMainAttributes().putValue("MixinConfigs", String.join(",", this.neoforgeMixins));
+            }
+
+            remapResources(forgeTemps, neoforgeTemps, fabricTemps, quiltTemps);
 
             if (this.forgeMixins != null && mergedManifest.getMainAttributes().getValue("MixinConfigs") == null) {
                 logger.debug("Couldn't detect forge mixins. You can ignore this if you are not using mixins with forge.\n" +
@@ -247,9 +280,20 @@ public class Forgix {
                 }
             }
 
+            if (this.neoforgeMixins != null && mergedManifest.getMainAttributes().getValue("MixinConfigs") == null) {
+                logger.debug("Couldn't detect neoforge mixins. You can ignore this if you are not using mixins with neoforge.\n" +
+                        "If this is an issue then you can configure mixins manually\n" +
+                        "Though we'll try to detect them automatically.\n");
+                if (!neoforgeMixins.isEmpty()) {
+                    logger.debug("Detected neoforge mixins: " + String.join(",", this.neoforgeMixins) + "\n");
+                    mergedManifest.getMainAttributes().putValue("MixinConfigs", String.join(",", this.neoforgeMixins));
+                }
+            }
+
             mergedManifest.getMainAttributes().putValue(manifestVersionKey, version);
 
             if (forgeJar != null && forgeJar.exists()) new File(forgeTemps, "META-INF/MANIFEST.MF").delete();
+            if (neoforgeJar != null && neoforgeJar.exists()) new File(neoforgeTemps, "META-INF/MANIFEST.MF").delete();
             if (fabricJar != null && fabricJar.exists()) new File(fabricTemps, "META-INF/MANIFEST.MF").delete();
             if (quiltJar != null && quiltJar.exists()) new File(quiltTemps, "META-INF/MANIFEST.MF").delete();
 
@@ -265,6 +309,7 @@ public class Forgix {
             outputStream.close();
 
             if (forgeJar != null && forgeJar.exists()) FileUtils.copyDirectory(forgeTemps, mergedTemps);
+            if (neoforgeJar != null && neoforgeJar.exists()) FileUtils.copyDirectory(neoforgeTemps, mergedTemps);
             if (fabricJar != null && fabricJar.exists()) FileUtils.copyDirectory(fabricTemps, mergedTemps);
             if (quiltJar != null && quiltJar.exists()) FileUtils.copyDirectory(quiltTemps, mergedTemps);
 
@@ -301,6 +346,10 @@ public class Forgix {
             if (forgeJar != null && forgeJar.exists()) {
                 FileUtils.deleteQuietly(forgeTemps);
                 forgeJar.delete();
+            }
+            if (neoforgeJar != null && neoforgeJar.exists()) {
+                FileUtils.deleteQuietly(neoforgeTemps);
+                neoforgeJar.delete();
             }
             if (fabricJar != null && fabricJar.exists()) {
                 FileUtils.deleteQuietly(fabricTemps);
@@ -366,6 +415,42 @@ public class Forgix {
                 forgeRelocator.run();
 
                 forgeJar = remappedForgeJar;
+            }
+
+            if (neoforgeJar != null && neoforgeJar.exists()) {
+                File remappedNeoForgeJar = new File(tempDir, "tempNeoForgeInMerging.jar");
+                if (remappedNeoForgeJar.exists()) remappedNeoForgeJar.delete();
+                remappedNeoForgeJar.createNewFile();
+
+                List<Relocation> neoforgeRelocation = new ArrayList<>();
+                neoforgeRelocation.add(new Relocation(group, "neoforge." + group));
+                if (neoforgeRelocations != null)
+                    neoforgeRelocation.addAll(neoforgeRelocations.entrySet().stream().map(entry -> new Relocation(entry.getKey(), entry.getValue())).collect(ArrayList::new, ArrayList::add, ArrayList::addAll));
+
+                AtomicReference<String> architectury = new AtomicReference<>();
+                architectury.set(null);
+
+                JarFile jarFile = new JarFile(neoforgeJar);
+                jarFile.stream().forEach(jarEntry -> {
+                    if (jarEntry.isDirectory()) {
+                        if (jarEntry.getName().startsWith("architectury_inject")) {
+                            architectury.set(jarEntry.getName());
+                        }
+                    } else {
+                        String firstDirectory = getFirstDirectory(jarEntry.getName());
+                        if (firstDirectory.startsWith("architectury_inject")) {
+                            architectury.set(firstDirectory);
+                        }
+                    }
+                });
+                jarFile.close();
+
+                if (architectury.get() != null) neoforgeRelocation.add(new Relocation(architectury.get(), "neoforge." + architectury.get()));
+
+                JarRelocator neoforgeJarRelocator = new JarRelocator(neoforgeJar, remappedNeoForgeJar, neoforgeRelocation);
+                neoforgeJarRelocator.run();
+
+                neoforgeJar = remappedNeoForgeJar;
             }
 
             if (fabricJar != null && fabricJar.exists()) {
@@ -481,15 +566,18 @@ public class Forgix {
         /**
          * This is the second remapping method
          * This basically remaps all resources such as mixins, manifestJars, etc.
-         * This method also finds all the forge mixins for you if not detected
+         * This method also finds all the forge/neoforge mixins for you if not detected
          *
          * @param forgeTemps  The extracted forge jar directory
+         * @param neoforgeTemps The extracted neoforge jar directory
          * @param fabricTemps The extracted fabric jar directory
          * @param quiltTemps  The extracted quilt jar directory
          * @throws IOException If something went wrong
          */
-        private void remapResources(File forgeTemps, File fabricTemps, File quiltTemps) throws IOException {
+        private void remapResources(File forgeTemps, File neoforgeTemps, File fabricTemps, File quiltTemps) throws IOException {
             if (forgeRelocations == null) forgeRelocations = new HashMap<>();
+            if (neoforgeRelocations == null) neoforgeRelocations = new HashMap<>();
+
             if (forgeJar != null && forgeJar.exists()) {
                 for (File file : manifestJars(forgeTemps)) {
                     File remappedFile = new File(file.getParentFile(), "forge-" + file.getName());
@@ -528,6 +616,58 @@ public class Forgix {
                     while (scanner.hasNext()) {
                         String line = scanner.nextLine();
                         for (Map.Entry<String, String> entry : forgeRelocations.entrySet()) {
+                            line = line.replace(entry.getKey(), entry.getValue());
+                        }
+                        sb.append(line).append("\n");
+                    }
+
+                    scanner.close();
+                    fis.close();
+                    FileOutputStream fos = new FileOutputStream(file);
+                    fos.write(sb.toString().getBytes());
+                    fos.flush();
+                    fos.close();
+                }
+            }
+
+            if (neoforgeJar != null && neoforgeJar.exists()) {
+                for (File file : manifestJars(neoforgeTemps)) {
+                    File remappedFile = new File(file.getParentFile(), "neoforge-" + file.getName());
+                    neoforgeRelocations.put(file.getName(), remappedFile.getName());
+                    file.renameTo(remappedFile);
+                }
+
+                for (File file : listAllPlatformServices(neoforgeTemps, group)) {
+                    File remappedFile = new File(file.getParentFile(), "neoforge." + file.getName());
+                    neoforgeRelocations.put(file.getName(), remappedFile.getName());
+                    file.renameTo(remappedFile);
+                }
+
+                neoforgeMixins = new ArrayList<>();
+                for (File file : listAllMixins(neoforgeTemps, false)) {
+                    File remappedFile = new File(file.getParentFile(), "neoforge-" + file.getName());
+                    neoforgeRelocations.put(file.getName(), remappedFile.getName());
+                    file.renameTo(remappedFile);
+
+                    neoforgeMixins.add(remappedFile.getName());
+                }
+
+                for (File file : listAllRefmaps(neoforgeTemps)) {
+                    File remappedFile = new File(file.getParentFile(), "neoforge-" + file.getName());
+                    neoforgeRelocations.put(file.getName(), remappedFile.getName());
+                    file.renameTo(remappedFile);
+                }
+
+                neoforgeRelocations.put(group, "neoforge." + group);
+                neoforgeRelocations.put(group.replace(".", "/"), "neoforge/" + group.replace(".", "/"));
+                for (File file : listAllTextFiles(neoforgeTemps)) {
+                    FileInputStream fis = new FileInputStream(file);
+                    Scanner scanner = new Scanner(fis);
+                    StringBuilder sb = new StringBuilder();
+
+                    while (scanner.hasNext()) {
+                        String line = scanner.nextLine();
+                        for (Map.Entry<String, String> entry : neoforgeRelocations.entrySet()) {
                             line = line.replace(entry.getKey(), entry.getValue());
                         }
                         sb.append(line).append("\n");
@@ -711,6 +851,11 @@ public class Forgix {
                     if (forgeJar != null && forgeJar.exists()) {
                         removeDuplicateRelocations.put("forge." + duplicate, duplicate);
                         removeDuplicateRelocationResources.put("forge/" + duplicatePath, duplicatePath);
+                    }
+
+                    if (neoforgeJar != null && neoforgeJar.exists()) {
+                        removeDuplicateRelocations.put("neoforge." + duplicate, duplicate);
+                        removeDuplicateRelocationResources.put("neoforge/" + duplicatePath, duplicatePath);
                     }
 
                     if (fabricJar != null && fabricJar.exists()) {

--- a/src/main/java/io/github/pacifistmc/forgix/plugin/ForgixMergeExtension.java
+++ b/src/main/java/io/github/pacifistmc/forgix/plugin/ForgixMergeExtension.java
@@ -83,6 +83,23 @@ public class ForgixMergeExtension {
         return forgeContainer;
     }
 
+    NeoForgeContainer neoForgeContainer;
+
+    public NeoForgeContainer neoforge(Closure<NeoForgeContainer> closure) {
+        neoForgeContainer = new NeoForgeContainer();
+        ForgixPlugin.rootProject.configure(neoForgeContainer, closure);
+        return neoForgeContainer;
+    }
+
+    public void setNeoForgeContainer(NeoForgeContainer neoForgeContainer) {
+        this.neoForgeContainer = neoForgeContainer;
+    }
+
+    public NeoForgeContainer getNeoForgeContainer() {
+        if (neoForgeContainer == null) neoForgeContainer = new NeoForgeContainer();
+        return neoForgeContainer;
+    }
+
     FabricContainer fabricContainer;
 
     public FabricContainer fabric(Closure<FabricContainer> closure) {
@@ -151,6 +168,48 @@ public class ForgixMergeExtension {
     @SuppressWarnings("InnerClassMayBeStatic")
     public class ForgeContainer {
         String projectName = "forge";
+        String jarLocation;
+        Map<String, String> additionalRelocates;
+        List<String> mixins;
+
+        public String getProjectName() {
+            return projectName;
+        }
+
+        public void setProjectName(String projectName) {
+            this.projectName = projectName;
+        }
+
+        public String getJarLocation() {
+            return jarLocation;
+        }
+
+        public void setJarLocation(String jarLocation) {
+            this.jarLocation = jarLocation;
+        }
+
+        public Map<String, String> getAdditionalRelocates() {
+            return additionalRelocates;
+        }
+
+        public void additionalRelocate(String from, String to) {
+            if (this.additionalRelocates == null) this.additionalRelocates = new java.util.HashMap<>();
+            this.additionalRelocates.put(from, to);
+        }
+
+        public List<String> getMixins() {
+            return mixins;
+        }
+
+        public void mixin(String mixin) {
+            if (this.mixins == null) this.mixins = new java.util.ArrayList<>();
+            this.mixins.add(mixin);
+        }
+    }
+
+    @SuppressWarnings("InnerClassMayBeStatic")
+    public class NeoForgeContainer {
+        String projectName = "neoforge";
         String jarLocation;
         Map<String, String> additionalRelocates;
         List<String> mixins;

--- a/src/main/java/io/github/pacifistmc/forgix/plugin/ForgixPlugin.java
+++ b/src/main/java/io/github/pacifistmc/forgix/plugin/ForgixPlugin.java
@@ -15,7 +15,7 @@ public class ForgixPlugin implements Plugin<Project> {
         settings = rootProject.getExtensions().create("forgix", ForgixMergeExtension.class);
         rootProject.getTasks().register("mergeJars", MergeJarsTask.class).configure(forgix -> {
             forgix.setGroup("forgix");
-            forgix.setDescription("Merges Fabric (also Quilt) and Forge jars into a single jar!");
+            forgix.setDescription("Merges Fabric (also Quilt), Forge and NeoForge jars into a single jar!");
         });
     }
 }

--- a/src/main/java/io/github/pacifistmc/forgix/plugin/MergeJarsTask.java
+++ b/src/main/java/io/github/pacifistmc/forgix/plugin/MergeJarsTask.java
@@ -24,12 +24,14 @@ public class MergeJarsTask extends DefaultTask {
             return;
         }
         ForgixMergeExtension.ForgeContainer forgeSettings = ForgixPlugin.settings.getForgeContainer();
+        ForgixMergeExtension.NeoForgeContainer neoforgeSettings = ForgixPlugin.settings.getNeoForgeContainer();
         ForgixMergeExtension.FabricContainer fabricSettings = ForgixPlugin.settings.getFabricContainer();
         ForgixMergeExtension.QuiltContainer quiltSettings = ForgixPlugin.settings.getQuiltContainer();
 
         List<ForgixMergeExtension.CustomContainer> customSettingsList = ForgixPlugin.settings.getCustomContainers();
 
         Project forgeProject = null;
+        Project neoforgeProject = null;
         Project fabricProject = null;
         Project quiltProject = null;
 
@@ -40,6 +42,10 @@ public class MergeJarsTask extends DefaultTask {
             forgeProject = ForgixPlugin.rootProject.getAllprojects().stream().filter(p -> !p.getName().equals(ForgixPlugin.rootProject.getName())).filter(p -> p.getName().equalsIgnoreCase(forgeSettings.getProjectName())).findFirst().get();
             validation.add(true);
         } catch (NoSuchElementException ignored) { }
+        try {
+            neoforgeProject = ForgixPlugin.rootProject.getAllprojects().stream().filter(p -> !p.getName().equals(ForgixPlugin.rootProject.getName())).filter(p -> p.getName().equalsIgnoreCase(neoforgeSettings.getProjectName())).findFirst().get();
+            validation.add(true);
+        } catch (NoSuchElementException ignored) {}
         try {
             fabricProject = ForgixPlugin.rootProject.getAllprojects().stream().filter(p -> !p.getName().equals(ForgixPlugin.rootProject.getName())).filter(p -> p.getName().equalsIgnoreCase(fabricSettings.getProjectName())).findFirst().get();
             validation.add(true);
@@ -64,6 +70,7 @@ public class MergeJarsTask extends DefaultTask {
         validation.clear();
 
         File forgeJar = null;
+        File neoforgeJar = null;
         File fabricJar = null;
         File quiltJar = null;
 
@@ -80,6 +87,23 @@ public class MergeJarsTask extends DefaultTask {
                         if (file.getName().length() < i || i == 0) {
                             i = file.getName().length();
                             forgeJar = file;
+                        }
+                    }
+                }
+            }
+        }
+
+        if (neoforgeProject != null) {
+            if (neoforgeSettings.getJarLocation() != null) {
+                neoforgeJar = new File(neoforgeProject.getProjectDir(), neoforgeSettings.getJarLocation());
+            } else {
+                int i = 0;
+                for (File file : new File(neoforgeProject.getBuildDir(), "libs").listFiles()) {
+                    if (file.isDirectory()) continue;
+                    if (io.github.pacifistmc.forgix.utils.FileUtils.isZipFile(file)) {
+                        if (file.getName().length() < i || i == 0) {
+                            i = file.getName().length();
+                            neoforgeJar = file;
                         }
                     }
                 }
@@ -141,7 +165,7 @@ public class MergeJarsTask extends DefaultTask {
         if (mergedJar.exists()) FileUtils.forceDelete(mergedJar);
         if (!mergedJar.getParentFile().exists()) mergedJar.getParentFile().mkdirs();
 
-        Path tempMergedJarPath = new Forgix.Merge(forgeJar, forgeSettings.getAdditionalRelocates(), forgeSettings.getMixins(), fabricJar, fabricSettings.getAdditionalRelocates(), quiltJar, quiltSettings.getAdditionalRelocates(), customJars, ForgixPlugin.settings.getGroup(), new File(ForgixPlugin.rootProject.getRootDir(), ".gradle" + File.separator + "forgix"), ForgixPlugin.settings.getMergedJarName(), ForgixPlugin.settings.getRemoveDuplicates(), ForgixPlugin.rootProject.getLogger()).merge(false).toPath();
+        Path tempMergedJarPath = new Forgix.Merge(forgeJar, forgeSettings.getAdditionalRelocates(), forgeSettings.getMixins(), neoforgeJar, neoforgeSettings.getAdditionalRelocates(), neoforgeSettings.getMixins(), fabricJar, fabricSettings.getAdditionalRelocates(), quiltJar, quiltSettings.getAdditionalRelocates(), customJars, ForgixPlugin.settings.getGroup(), new File(ForgixPlugin.rootProject.getRootDir(), ".gradle" + File.separator + "forgix"), ForgixPlugin.settings.getMergedJarName(), ForgixPlugin.settings.getRemoveDuplicates(), ForgixPlugin.rootProject.getLogger()).merge(false).toPath();
         Files.move(tempMergedJarPath, mergedJar.toPath(), StandardCopyOption.REPLACE_EXISTING);
         try {
             Files.setPosixFilePermissions(mergedJar.toPath(), Forgix.Merge.perms);

--- a/src/main/java/io/github/pacifistmc/forgix/plugin/MergeJarsTask.java
+++ b/src/main/java/io/github/pacifistmc/forgix/plugin/MergeJarsTask.java
@@ -80,16 +80,7 @@ public class MergeJarsTask extends DefaultTask {
             if (forgeSettings.getJarLocation() != null) {
                 forgeJar = new File(forgeProject.getProjectDir(), forgeSettings.getJarLocation());
             } else {
-                int i = 0;
-                for (File file : new File(forgeProject.getBuildDir(), "libs").listFiles()) {
-                    if (file.isDirectory()) continue;
-                    if (io.github.pacifistmc.forgix.utils.FileUtils.isZipFile(file)) {
-                        if (file.getName().length() < i || i == 0) {
-                            i = file.getName().length();
-                            forgeJar = file;
-                        }
-                    }
-                }
+                forgeJar = io.github.pacifistmc.forgix.utils.FileUtils.findLatestFile(new File(forgeProject.getBuildDir(), "libs"));
             }
         }
 
@@ -97,16 +88,7 @@ public class MergeJarsTask extends DefaultTask {
             if (neoforgeSettings.getJarLocation() != null) {
                 neoforgeJar = new File(neoforgeProject.getProjectDir(), neoforgeSettings.getJarLocation());
             } else {
-                int i = 0;
-                for (File file : new File(neoforgeProject.getBuildDir(), "libs").listFiles()) {
-                    if (file.isDirectory()) continue;
-                    if (io.github.pacifistmc.forgix.utils.FileUtils.isZipFile(file)) {
-                        if (file.getName().length() < i || i == 0) {
-                            i = file.getName().length();
-                            neoforgeJar = file;
-                        }
-                    }
-                }
+                neoforgeJar = io.github.pacifistmc.forgix.utils.FileUtils.findLatestFile(new File(neoforgeProject.getBuildDir(), "libs"));
             }
         }
 
@@ -114,16 +96,7 @@ public class MergeJarsTask extends DefaultTask {
             if (fabricSettings.getJarLocation() != null) {
                 fabricJar = new File(fabricProject.getProjectDir(), fabricSettings.getJarLocation());
             } else {
-                int i = 0;
-                for (File file : new File(fabricProject.getBuildDir(), "libs").listFiles()) {
-                    if (file.isDirectory()) continue;
-                    if (io.github.pacifistmc.forgix.utils.FileUtils.isZipFile(file)) {
-                        if (file.getName().length() < i || i == 0) {
-                            i = file.getName().length();
-                            fabricJar = file;
-                        }
-                    }
-                }
+                fabricJar = io.github.pacifistmc.forgix.utils.FileUtils.findLatestFile(new File(fabricProject.getBuildDir(), "libs"));
             }
         }
 
@@ -131,16 +104,7 @@ public class MergeJarsTask extends DefaultTask {
             if (quiltSettings.getJarLocation() != null) {
                 quiltJar = new File(quiltProject.getProjectDir(), quiltSettings.getJarLocation());
             } else {
-                int i = 0;
-                for (File file : new File(quiltProject.getBuildDir(), "libs").listFiles()) {
-                    if (file.isDirectory()) continue;
-                    if (io.github.pacifistmc.forgix.utils.FileUtils.isZipFile(file)) {
-                        if (file.getName().length() < i || i == 0) {
-                            i = file.getName().length();
-                            quiltJar = file;
-                        }
-                    }
-                }
+                quiltJar = io.github.pacifistmc.forgix.utils.FileUtils.findLatestFile(new File(quiltProject.getBuildDir(), "libs"));
             }
         }
 
@@ -148,16 +112,7 @@ public class MergeJarsTask extends DefaultTask {
             if (entry.getValue().getJarLocation() != null) {
                 customJars.put(entry.getValue(), new File(entry.getKey().getProjectDir(), entry.getValue().getJarLocation()));
             } else {
-                int i = 0;
-                for (File file : new File(entry.getKey().getBuildDir(), "libs").listFiles()) {
-                    if (file.isDirectory()) continue;
-                    if (io.github.pacifistmc.forgix.utils.FileUtils.isZipFile(file)) {
-                        if (file.getName().length() < i || i == 0) {
-                            i = file.getName().length();
-                            customJars.put(entry.getValue(), file);
-                        }
-                    }
-                }
+                customJars.put(entry.getValue(), io.github.pacifistmc.forgix.utils.FileUtils.findLatestFile(new File(entry.getKey().getBuildDir(), "libs")));
             }
         }
 

--- a/src/main/java/io/github/pacifistmc/forgix/utils/FileUtils.java
+++ b/src/main/java/io/github/pacifistmc/forgix/utils/FileUtils.java
@@ -1,6 +1,7 @@
 package io.github.pacifistmc.forgix.utils;
 
 import org.apache.commons.io.FilenameUtils;
+import org.jetbrains.annotations.Nullable;
 
 import java.io.*;
 import java.nio.charset.Charset;
@@ -227,6 +228,45 @@ public class FileUtils {
         } catch (IOException exception) {
             return false;
         }
+    }
+
+    /**
+     * Try to find the latest file with the shortest name in a directory
+     * @param directory The directory to find the file in
+     * @return The newest file or null if no files were found
+     */
+    @Nullable
+    public static File findLatestFile(File directory) {
+        File latestFile = null;
+        long lastModifiedTime = Long.MIN_VALUE;
+        int i = 0;
+        List<File> processedFiles = new ArrayList<>();
+
+        File[] files = directory.listFiles();
+        if (files == null)
+            return null;
+
+        for (File f : files) {
+            if (f.isDirectory() || !isZipFile(f))
+                continue;
+
+            if (f.getName().contains("-shadow") || f.getName().contains("-sources") || f.getName().contains("-dev"))
+                continue;
+
+            if (f.lastModified() > lastModifiedTime) {
+                lastModifiedTime = f.lastModified();
+                processedFiles.add(f);
+            }
+        }
+
+        for (File f : processedFiles) {
+            if (f.getName().length() < i || i == 0) {
+                latestFile = f;
+                i = f.getName().length();
+            }
+        }
+
+        return latestFile;
     }
 
     /**

